app/views/finance/invoices/_invoices.html.haml

```diff
@@ -26,6 +26,10 @@
         %td= link_to Delivery.model_name.human, [invoice.supplier,invoice.delivery] if invoice.delivery
         %td= link_to format_date(invoice.order.ends), new_finance_order_path(order_id: invoice.order_id) if invoice.order
         %td= truncate(invoice.note)
-        %td= link_to t('ui.edit'), edit_finance_invoice_path(invoice), class: 'btn btn-mini'
-        %td= link_to t('ui.delete'), finance_invoice_path(invoice), :data => {:confirm => t('.confirm_delete')}, :method => :delete,
-          class: 'btn btn-danger btn-mini'
+        %td
+          - if current_user.role_finance? or (current_user.role_invoices? and not invoice.paid_on)
```

Comment:
Ok, my tension comes from this:
1. As an invoice-user, I create an invoice for something I bought for the foodcoop
2. Another invoice-user edits the invoice (and I'm not aware of it).
3. The finance people see the invoice accept it.
4. Later someone checks the invoices and finds an error.
5. I'm held accountable, since I created the invoice. (Or, if the finance people see something and contact me, I'm at loss why the invoice is different from what I entered before.)

This is a kind of worst-case scenario, but I'd like to avoid these things when possible. The occasional time that someone else needs to edit the invoice that I entered can go through communication with me.

Now perhaps you might work more in groups entering invoices, where the responsibility is shared. If that's so, I'd like it to be clear in the interface that the group is responsible (and others than the group can't edit it).

That's why I'd like to restrict editing to your own invoice. Or implement a system where it's clear who makes changes.